runtime/j9vm/java11vmi.c

```diff
@@ -730,6 +756,28 @@ JVM_DefineModule(JNIEnv * env, jobject module, jboolean isOpen, jstring version,
 		J9ClassLoader * const classLoader = getModuleObjectClassLoader(currentThread, modObj);
 		j9object_t moduleName = J9VMJAVALANGMODULE_NAME(currentThread, modObj);
 
+#if JAVA_SPEC_VERSION >= 15
+		if (NULL == packages) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto done;
+		} else {
+			jsize pkgIndex = 0;
+			memset(packages, 0, packagesNumBytes);
+			for (pkgIndex = 0; pkgIndex < numPackages; pkgIndex++) {
+				jobject packageObj = (*env)->GetObjectArrayElement(env, packageArray, pkgIndex);
```

Comment:
@gacholio Is this legal? Calling JNI functions _after_ `internalEnterVMFromJNI()` and acquiring `classLoaderModuleAndLocationMutex`?

Reply:
Absolutely not.